### PR TITLE
feat: show deposit/voting period from gov params and add alert

### PR DESCRIPTION
**Pull request by poomthiti**
Show deposit/voting period from gov params and add minimum required alert

### CODE REVIEW

**Review by bkioshn — APPROVED** [review]

File: src/lib/services/delegation.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export const getStakingParams = async (
     `${endpoint}/cosmos/staking/v1beta1/params`
   );
   return {
-    unbondingTime: secondsToDays(data.params.unbonding_time.slice(0, -1)),
+    unbondingTime: formatSeconds(data.params.unbonding_time.slice(0, -1)),
```
Comment (evilpeach) [inline src/lib/services/delegation.ts:123]: remove `slice`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/services/proposal.ts [hunk 2]
```diff
@@ -17,6 +17,24 @@ export const fetchGovDepositParams = (
...
+interface ProposalVotingPeriod {
+  proposal_type: string;
+  voting_period: string;
+}
+
+export interface VotingParams {
+  voting_period: string;
+  proposal_voting_periods: ProposalVotingPeriod[];
+  expedited_voting_period: string;
+}
```
Comment (evilpeach) [inline src/lib/services/proposal.ts:29]: Let's use `SnakeToCamelCase` type condition and use snake case for raw data from lcd and camel case for internal use

Ref. https://www.typescriptlang.org/play?#code/C4TwDgpgBAygdgQwNYQCoHsDCCC2EA22AzhADwxQQAewEcAJkVEcAE4CWcA5gHxQC8sSjTqMoAAwAkAb04AzCKyioAvgH0Z8xVACqK8QCgoUAPwSZqmdjDtgCfOwBeZeMjRZcBYmR08e+oygALlgAbgMDUEgoKgFYRBQMbDxCBBJSAHIcTjVOW3Z7NXoIMHQiWzUAJgyeCMjwaFdEjxTvADkIFgh6UlQ+QVRhWgYmdAAjACsIAGNgQLNpQONjAG0AaShOKBQQdDllKDT4tyTPVPSNgDJmNk5eAF0QpvdkrzSIDq6epeXjVHX7j9jDxwr8VIEQqhwgZOLRWHIENNoAAlBAAdwAIiUyrYAAoIVi4JiLAD0JOW2TgRWx5WAITgAFccGNFCt7qCoDgEFRqaVaWpIBx0PQQiwONxwmSKTlqJB6LZurycXSoIzmaz2YFKbk4PlCsU+RVCcB2OhRbcJYFSugcGo5OxWCwQotfvbHcA1GN8OhhebxVwOcZ6OgGV6IGokOx8PhnUDlsHKQgTegqXgEFSEAA3LghMY+-AQdOB35QLjChwoNRw22R6OxksN7ZpehqNITegTJCtogdhD0pks1hx4zgsGBcHg+rRLGG4D4wk4JiCZ6nVrvT60HqozE0vEEom1IA

File: src/lib/services/proposalService.ts [hunk 3]
```diff
@@ -274,20 +279,24 @@ interface DepositParamsReturn extends Omit<DepositParams, "min_deposit"> {
     formattedDenom: string;
     formattedToken: string;
   };
+  initialDeposit: Token;
```
Comment (evilpeach) [inline src/lib/services/proposalService.ts:282]: let's rename to `minInitialDeposit`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/services/proposal.ts [hunk 4]
```diff
@@ -2,20 +2,51 @@ import type { Coin } from "@cosmjs/stargate";
 import axios from "axios";
 
 import type { AccessConfigPermission, Addr } from "lib/types";
+import type { SnakeToCamelCaseNested } from "lib/types/converter";
```
Comment (evilpeach) [inline src/lib/services/proposal.ts:5]: This type should be imported from `lib/types`

File: src/lib/services/proposal.ts [hunk 5]
```diff
@@ -2,20 +2,51 @@ import type { Coin } from "@cosmjs/stargate";
...
+  expedited_voting_period: string;
+}
+
+export type VotingParamsInternal = SnakeToCamelCaseNested<VotingParams>;
+
+export const fetchGovVotingParams = (
+  lcdEndpoint: string
+): Promise<VotingParamsInternal> =>
+  axios
+    .get(`${lcdEndpoint}/cosmos/gov/v1beta1/params/voting`)
```
Comment (evilpeach) [inline src/lib/services/proposal.ts:43]: `get<{voting_params: VotingParams}>`

**Review by evilpeach — APPROVED** [review]
lgtm

**Pull request merged**